packages/generators/app/src/index.ts

```diff
@@ -47,50 +43,50 @@ export const generateNewApp = (projectDirectory: string, options: Partial<NewOpt
     docker: process.env.DOCKER === 'true',
     deviceId: machineID(),
     tmpPath,
-    // use yarn if available and --use-npm isn't true
-    useYarn: !useNpm && hasYarn(),
     installDependencies: true,
-    strapiDependencies: [
-      '@strapi/strapi',
-      '@strapi/plugin-users-permissions',
-      '@strapi/plugin-cloud',
-    ],
-    additionalsDependencies: {
+    devDependencies: {},
+    dependencies: {
+      '@strapi/strapi': packageJson.version,
+      '@strapi/plugin-users-permissions': packageJson.version,
+      '@strapi/plugin-cloud': packageJson.version,
+      // third party
       react: '^18.0.0',
       'react-dom': '^18.0.0',
       'react-router-dom': '^6.0.0',
       'styled-components': '^6.0.0',
     },
-    useTypescript: Boolean(options.typescript),
   };
 
-  sentry.configureScope(function configureScope(sentryScope) {
-    const tags = {
-      os: os.type(),
-      osPlatform: os.platform(),
-      osArch: os.arch(),
-      osRelease: os.release(),
-      version: scope.strapiVersion,
-      nodeVersion: process.versions.node,
-      docker: scope.docker,
+  if (scope.useTypescript) {
+    scope.devDependencies = {
+      ...scope.devDependencies,
+      typescript: '^5',
+      '@types/node': '^20',
+      '@types/react': '^18',
+      '@types/react-dom': '^18',
     };
+  }
 
-    (Object.keys(tags) as Array<keyof typeof tags>).forEach((tag) => {
-      sentryScope.setTag(tag, tags[tag]);
-    });
-  });
+  addDatabaseDependencies(scope);
 
-  parseDatabaseArguments({ scope, args: options });
   initCancelCatcher();
 
-  return generateNew(scope).catch((error) => {
-    console.error(error);
-    return captureException(error).then(() => {
-      return trackError({ scope, error }).then(() => {
-        process.exit(1);
-      });
+  try {
+    await trackUsage({ event: 'willCreateProject', scope });
+
+    // create a project with full list of questions
+    return await createProject(scope);
+  } catch (error) {
+    if (!(error instanceof Error)) {
+      throw error;
+    }
+
+    console.log(`\n${error.message}\n`);
```

Comment:
If we don't want to ignore errors thrown as strings, we could do the following

```suggestion
    let e: Error;
 
    if (typeof error === 'string') {
        e = new Error(e);
    }
    else if (error instanceof Error) {
        e = error;
    }
    else {
        throw error
    }
    
    console.log(`\n${e.message}\n`);
```
but that might be a bit too much, as you prefer

Reply:
If you throw a string you are a bad person :D more seriously it means those are probably not our own errors so I let thme bubble up.

Reply:
make sense